enhancements/kueue/openshift-kueue.md

```diff
@@ -0,0 +1,749 @@
+Kueue with multiple replicas.
+
+Kueue webhooks are related to a service so one can monitor the webhooks service
+to determine if the webhooks are done.
+
+##### Machines
+
+Kueue will only create machines in the context of Provisioning Requests.
+There is work ongoing to implement Provising Requests in Hypershift.
+Once this is complete, Kueue can use this for autoscaling.
+
+We do not foresee any issue with integration with HCP.
+Kueue does provide webhooks on pods and that could cause some 
+performance issues in HCP. 
+We will do performance testing on HCP. 
+
+Kueue is an upstream project and its APIs are not aware of MachineSets.
+
+The one area that Kueue can create new machines is its use of Provision requests.
+There will be work to make sure HCP works with ProvisionRequests.
+
+##### Infra / Worker Nodes
+
+There should really be no issue with deploying Kueue on Infra or worker nodes.
+Kueue is agnostic to this.
+
+##### Metrics
+
+Autoscaling, based on metrics, may be tricky. Kueue will deploy its own metrics 
+on the customer workload nodes.
+The control plane will not be able to react to these metrics.
+
+##### APIServices
+
+APIService allow one to set different priorities for apiserver.
+Kueue has a feature called [VisibilityOnDemand](https://kueue.sigs.k8s.io/docs/tasks/manage/monitor_pending_workloads/pending_workloads_on_demand/#configure-api-priority-and-fairness) that requires one to deploy
+an apiservice.
+
+There are [outstanding security issues](https://github.com/kubernetes-sigs/kueue/issues/4433) for this feature so we are disabling this feature in our operator.
+
+#### Standalone Clusters
+
+Yes. Standalone clusters will be able to install Kueue from OperatorHub.
+
+#### Single-node Deployments or MicroShift
+
+##### Microshift
+
+To support microshift, one would need to install OLM.
+Once OLM is enabled, one can install Cert Manager and the operator.
+After that, the operator will also require that metrics be disabled. We will include a special flag for rare cases 
+where customers would disable metrics.
+
+Microshift does not enable openshift-monitoring so metrics would be disabled in this case.
+
+##### Single-node Deployments
+
+I don't forsee any issue with SNO.
+
+### Implementation Details/Notes/Constraints
+
+Cert Manager will be used to manage certificates so our operator will have a hard dependency on Cert Manager.
+
+#### Release Schedule
+
+| Kueue Operator     |  Stage       |  OCP Version   |  Kueue   | RHOAI GA
+| ------------------ | -------      | ---------------| -------- | -------
+| 0.1                | Tech Preview | 4.19           | 0.11.z   | N
+| 0.1                | Tech Preview | 4.20           | 0.13.z   | Y
+| ?                  | GA           | 4.19           | ?        | Y
+
+Kueue releases 6 times a year, roughly. 
+They will have roughly 2 releases per k8s version so we can take the latest version that is built with the kubernetes version that OCP comes with.
```

Comment:
Following up from the previous comment - we probably should also chart out release schedule for Kueue operator and map Kueue <-> Kueue operator versions?

Reply:
We have three phases for this and right now I am working on getting a tech preview out that would focus on a single kueue version to operator. 

Phase 2 would be RHOAI integration where we figure out versioning.

Right now, 0.1 of the operator would be using Kueue 0.10 or 0.11 depending on when we get everything signed off for Kueue. We are still waiting on production tasks to official release Kueue.